### PR TITLE
Fix security schema in API design

### PR DESCRIPTION
**Pull request by jraddaoui**
Use Goa's JWTSecurity, the API is only verifying the token against the OIDC provider, it's not actually doing the authentication flow.

Refs #957.

### CODE REVIEW

File: dashboard/src/openapi-generator/models/Location.ts
Comment (djjuhasz) [inline dashboard/src/openapi-generator/models/Location.ts:1]: The changes to this file are all adding unnecessary whitespace. :(  Maybe it's from using a different version of the generator?

**Review by djjuhasz — APPROVED** [review]
Mostly LGTM - I just noticed one of the re-generated files added spurious whitespace. It's not a blocker, but it would be nice to remove the whitespace.

File: dashboard/src/openapi-generator/models/Location.ts
Comment (jraddaoui) [reply dashboard/src/openapi-generator/models/Location.ts:1]: I used `make gen-dashboard-client`, which has `openapitools/openapi-generator-cli:v7.1.0` hardcoded in it, and each time I run it is the same generated code.
